src/internal/m365/exchange/events_restore.go

```diff
@@ -102,8 +109,181 @@ func (h eventRestoreHandler) restore(
 		return nil, clues.Stack(err)
 	}
 
+	// Have to parse event again as we modified the original event and
+	// removed cancelled and exceptions events form it
+	event, err = api.BytesToEventable(body)
+	if err != nil {
+		return nil, clues.Wrap(err, "creating event from bytes").WithClues(ctx)
+	}
```

Comment:
Any thoughts on how/if we can avoid reading it again? This is necessary due to the call to `toEventSimplified`.

Reply:
How concerned are we about re-reading it in the first place?  I assume it's a pretty small size, and not likely to consume many resources.  Working around that sounds like premature optimization.